cmd/web.go

```diff
@@ -585,12 +585,12 @@ func runWeb(ctx *cli.Context) error {
 			m.Get("/src/*", repo.Home)
 			m.Get("/raw/*", repo.SingleDownload)
 			m.Get("/commits/*", repo.RefCommits)
-			m.Get("/commit/:sha([a-z0-9]{7,40})$", repo.Diff)
+			m.Get("/commit/:sha([a-z0-9]{40})$", repo.Diff)
 			m.Get("/forks", repo.Forks)
 		}, context.RepoRef())
-		m.Get("/commit/:sha([a-z0-9]{7,40})\\.:ext(patch|diff)", repo.RawDiff)
+		m.Get("/commit/:sha([a-z0-9]{40})\\.:ext(patch|diff)", repo.RawDiff)
```

Comment:
This modification don't fix your issue. But the regexp is not correct : SHA1 only accept _a b c d e f 0 1 2 3 4 5 6 7 8 9_ characters. Could you replace `:sha([a-z0-9]{7,40})` with `:sha([a-f0-9]{7,40})` please.